src/relay/libp2p_relay_server.erl

```diff
@@ -269,8 +269,9 @@ shuffle(List) ->
 
 %% merge new peers into old peers based on their address
 merge_peers(NewPeers, OldPeers) ->
-    maps:values(maps:merge(maps:from_list([{libp2p_peer:pubkey_bin(P), P} || P <- OldPeers]),
-                           maps:from_list([{libp2p_peer:pubkey_bin(P), P} || P <- NewPeers]))).
+    Peers = maps:values(maps:merge(maps:from_list([{libp2p_peer:pubkey_bin(P), P} || P <- OldPeers]),
```

Comment:
This is the end goal to remove stale peers from relay_server, because we see a lot of them (relay server) having 600+ peers in state, which normally should be impossible but somehow happening so my guess is there are a lot of stale peers in this state.

Last resort would be to not use notify and just grab peers from peerbook everytime